frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -303,17 +326,43 @@ export default ({
         }
       })
     },
-    setInitMessages () {
-      if (this.isJoinedChatRoom(this.currentChatRoomId)) {
-        this.messages = cloneDeep(this.chatRoomLatestMessages)
+    getSimulatedState (initialize = true) {
+      return {
+        settings: cloneDeep(this.chatRoomSettings),
+        attributes: cloneDeep(this.chatRoomAttributes),
+        users: cloneDeep(this.chatRoomUsers),
+        messages: initialize ? [] : this.messages,
+        simulation: true
+      }
```

Comment:
This seems a bit repetitive, wouldn't it be easier to first clone `currentChatRoomState` and then update the individual properties that differ?

Reply:
Nice catch. Need to be fixed.